Bug: #[routes] macro generates Depends::resolve() requiring Injectable bound for factory types

## Description

The `#[routes]` macro generates `Depends::<T>::resolve()` for `#[inject]` parameters. `resolve()` requires `T: Injectable` to support the fallback path (`T::inject()` for types not registered in the global DI registry). However, types produced by `#[injectable_factory]` do not implement `Injectable` — they rely solely on registry-based resolution.

This causes compilation errors when upgrading past the commit where `Injectable` bound was added to `Depends::resolve()`:

```
error[E0277]: the trait bound `DashboardRouter: Injectable` is not satisfied
```

## Root Cause

`Depends` has two resolution methods:

| Method | `Injectable` bound | Fallback | Purpose |
|--------|-------------------|----------|---------|
| `resolve()` | Required | Registry → `T::inject()` | Types with manual `Injectable` impl |
| `resolve_from_registry()` | Not required | Registry only | Factory-produced types |

The `#[routes]` macro always generates `resolve()` (line 251 in `routes_registration.rs`) to support both patterns. However, when `T` is a factory-produced type that does not implement `Injectable`, the `Injectable` bound fails at compile time — even though the `inject()` fallback would never be reached at runtime.

## Steps to Reproduce

```rust
#[injectable_factory(scope = "transient")]
async fn make_router(...) -> DashboardRouter { ... }

#[routes]
pub async fn routes(#[inject] router: Depends<DashboardRouter>) -> UnifiedRouter {
    router.try_unwrap().unwrap().0
}
```

## Possible Solutions

1. **Remove `Injectable` bound from `resolve()`** — Use trait-based dispatch (e.g., specialization or separate trait) to optionally invoke `inject()` when available
2. **Add macro attribute to select resolution method** — e.g., `#[inject(registry_only)]` to generate `resolve_from_registry()` instead of `resolve()`
3. **Generate `resolve_from_registry()` by default in `#[routes]`** — Since `#[routes]` functions typically receive factory-produced types via DI, registry-only resolution is more appropriate as the default

## Related

- PR #3517: Introduced `resolve_from_registry()` and added `Injectable` bound to `resolve()`
- PR #3509: Deprecated `Injected<T>` in favor of `Depends<T>`

## Environment

- reinhardt-web: 0.1.0-rc.15 (main branch, commit 5a90255a)
- Rust edition: 2024

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-core/macros/src/routes_registration.rs
+++ b/crates/reinhardt-core/macros/src/routes_registration.rs
@@ -246,9 +246,11 @@ pub(crate) fn routes_impl(_args: TokenStream, input: ItemFn) -> Result<TokenStre
 			.iter()
 			.map(|(pat, ty)| {
 				if let Some(inner_ty) = extract_depends_inner_type(ty) {
-					// Parameter is Depends<T>: resolve via Depends::resolve()
+					// Parameter is Depends<T>: resolve via registry only.
+					// Factory-produced types (via #[injectable_factory]) do not implement
+					// Injectable, so resolve_from_registry() is used to avoid requiring the bound.
 					quote! {
-						let #pat: #ty = #di_crate::Depends::<#inner_ty>::resolve(&*__ctx, true).await
+						let #pat: #ty = #di_crate::Depends::<#inner_ty>::resolve_from_registry(&*__ctx, true).await
 							.map_err(|e| -> ::std::boxed::Box<dyn ::std::error::Error + Send + Sync> {
 								::std::boxed::Box::new(e)
 							})?;

